tests/datasource/fluent/test_snowflake_datasource.py

```diff
@@ -100,7 +100,9 @@ def test_conflicting_connection_string_and_args_raises_error(
     connection_string: ConfigStr | SnowflakeDsn | None | dict, connect_args: dict
 ):
     with pytest.raises(ValueError):
-        _ = SnowflakeDatasource(connection_string=connection_string, **connect_args)
+        _ = SnowflakeDatasource(
+            name="my_sf_ds", connection_string=connection_string, **connect_args
+        )
```

Comment:
This test wasn't working properly. It was passing only because same error was being thrown for missing name, and when this was fixed, passing because improper url was also throwing same error. 

When test was fixed, it revealed that validation of connection_string XOR connection args wasn't happening.